website/marietje/models.py

```diff
@@ -0,0 +1,179 @@
+    def configured(self):
+        """
+        Check if this object is ready to play music.
+
+        :return: True if this object is ready for music playback, False otherwise
+        """
+        return self.playback_device_id is not None
+
+    @property
+    def currently_playing(self):
+        """
+        Get currently playing music information.
+
+        :return: a dictionary with the following content:
+        {
+            image: [link to image of track],
+            name: [name of currently playing track],
+            artists: [list of artist names],
+            is_playing: [True|False]
+        }
+        """
+        if not self.configured:
+            raise RuntimeError("This Spotify settings object is not configured yet.")
+
+        currently_playing = self.spotify.currently_playing()
+
+        if currently_playing is None:
+            return False
+
+        image = currently_playing["item"]["album"]["images"][0]["url"]
+        name = currently_playing["item"]["name"]
+        artists = [x["name"] for x in currently_playing["item"]["artists"]]
+
+        return {
+            "image": image,
+            "name": name,
+            "artists": artists,
+            "is_playing": currently_playing["is_playing"],
+        }
+
+    @property
+    def cache_path(self):
+        """
+        Get the cache file path for this auth object.
+
+        :return: the cache file path
+        """
+        return os.path.join(settings.SPOTIFY_CACHE_PATH, self.client_id)
+
+    @property
+    def auth(self):
+        """
+        Get a SpotifyOAuth object from this database object.
+
+        :return: a SpotifyOAuth object
+        """
+        return SpotifyOAuth(
+            client_id=self.client_id,
+            client_secret=self.client_secret,
+            redirect_uri=self.redirect_uri,
+            cache_path=self.cache_path,
+            scope=self.SCOPE,
+        )
+
+    @property
+    def spotify(self):
+        """
+        Get a Spotify object with a SpotifyOAuth manager as authentication backend.
+
+        :return: a Spotify object
+        """
+        return Spotify(oauth_manager=self.auth)
+
+    @property
+    def get_display_name(self):
+        """
+        Get the display name of the user.
+
+        :return: the display name of the user authenticated
+        """
+        if self.display_name is None:
+            self.display_name = self.spotify.me()["display_name"]
+            self.save()
+        return self.display_name
+
+    def __str__(self):
+        """
+        Convert this object to string.
+
+        :return: the display name if it is not None, the client id otherwise
+        """
+        if self.display_name is not None:
+            return self.display_name
+        else:
+            return self.client_id
+
+    class Meta:
+        """Meta class."""
+
+        verbose_name = "Spotify settings"
+        verbose_name_plural = "Spotify settings"
+
+
+class SpotifyArtist(models.Model):
```

Comment:
Is this model really required

Reply:
It might be nice for some statistics later on

Reply:
Okidoki